x-pack/libbeat/management/managerV2.go

```diff
@@ -669,6 +673,10 @@ func (cm *BeatV2Manager) reload(units map[unitKey]*agentUnit) {
 		cm.logger.Errorw("setting output state", "error", err)
 	}
 
+	// reload APM tracing configuration
+	// all error handling is handled inside of reloadAPM
```

Comment:
Both `reloadInputs` and `reloadOutput` return an error, why does `reloadAPM` not?

Reply:
Because in no case, do we want APM trace configuration to cause the beat to fail. I don't think we want that, so I handle all of that logic internally inside of `reloadAPM`.

Reply:
I updated the docstring on `reloadAPM` to provide this reasoning.